app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -218,17 +222,36 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         themeManager.applyStatusBarTheme(this)
     }
 
-    private fun setupToolbarAndNavigation() {
+    fun getSupportActionBarAndInflateIfNecessary(): Toolbar? {
         // Add ids to this that we don't want to have a toolbar back button
-        val appBarConfiguration = AppBarConfiguration.Builder().build()
-        val navigationToolbar = findViewById<Toolbar>(R.id.navigationToolbar)
-        setSupportActionBar(navigationToolbar)
-        NavigationUI.setupWithNavController(navigationToolbar, navHost.navController, appBarConfiguration)
-        navigationToolbar.setNavigationOnClickListener {
-            onBackPressed()
+        if (!this.toolbarHasInflated) {
+            val appBarConfiguration = AppBarConfiguration.Builder().build()
```

Comment:
super-nit: I think we can inline this function call: I don't see any value in leaving it out. It just makes this block more verbose.